### DISCUSSION TITLE
wiring of MAR3501 3.5" arduino uno TFT shield to raspberry pico

### DISCUSSION THREAD

**Discussion by fbeleznay** [Bodmer/TFT_eSPI#2743] [Compatible displays and setup files]
This is a beginner question. I have a MAR3501 3.5" arduino uno TFT shield (picture attached) and it looks to me that it uses analog pins to connect the LCD to the Uno (it uses the SPI pins to access the SD card).
The pico has less analog pins that this shield is using for the connection.
I looked at example codes and tutorials, but the ones I could find explain how to connect TFT display using SPI.
In [this video](https://www.youtube.com/watch?v=D3lv0eySz8A) the author shows a shield with similar pinout and at around 2:00 he states that the SPI connection is shared by the SD card and the LCD. Is it the same with the display I have? Is it enough to connect the SPI pins to be able to draw on the display and also access the SD card?
Can someone please point me to a place where I can find a wiring to connect the shield to the Pico and some example code to start my learning?
![Mar3501](https://github.com/Bodmer/TFT_eSPI/assets/111730066/9dd29267-2a3e-4b01-8c60-a19c0bee4476)

**Comment by Bodmer** [comment]
The display is 8 bit parallel. The SD card interface is SPI.

This setup file should work.
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setups/Setup103_RP2040_ILI9486_parallel.h

**Reply by fbeleznay** [reply]
Unfortunately, this did not work for me. I can see in https://github.com/Bodmer/TFT_eSPI/discussions/1740 that it works for others, but for some reason I only see a bright white screen as soon as I connect the Pico to my computer and this does not change no matter what demo program I upload.
Maybe it is because of my wiring, but let me upload some pictures just in case you notice any obvious mistakes.
Here is the setup I use:
![setup](https://github.com/Bodmer/TFT_eSPI/assets/111730066/82089a35-bd98-4c76-b76d-135ecb9d2dc4)
Here are some pictures of my wiring:
![wiring](https://github.com/Bodmer/TFT_eSPI/assets/111730066/e8d4cb15-53ab-4822-b8b6-3cba1a7ffdd2)
I had a problem with the instructions in the setup. The file talks about TFT_DC, but on my board there is no such reference. Based on the pinout diagram I found on the internet (and the pictures from the discussion I linked above), I connected pin 28 to LCD_RS on my board.
![pinuot](https://github.com/Bodmer/TFT_eSPI/assets/111730066/ecf7b663-57be-4b2b-9bf6-c5b4bcbd2082)
I check my connections in my wiring, but if you have any other suggestions, please, let me know.

Also, I ran the Read_User_Setup example, but I could not find output it mentions. Where can I find it? I am using Arduino IDE on Mac OS.

**Reply by Bodmer** [reply]
The DC signal is the same as RS so that is fine.

The output of the Read_User_Setup sketch is sent to the serial port and will be visible if you open the Arduino IDE serial monitor window. Set the baud rate to 115200 in the window. Check this output.

**Reply by fbeleznay** [reply]
I copy here the Read_user_Setup output.

[code]
TFT_eSPI ver = 2.5.0
Processor    = RP2040
Transactions = Yes
Interface    = Parallel
Display driver = 9486
Display width  = 320
Display height = 480

TFT_DC   = GPIO 28
TFT_RST  = GPIO 2
TFT_WR   = GPIO 22

TFT_D0   = GPIO 6
TFT_D1   = GPIO 7
TFT_D2   = GPIO 8
TFT_D3   = GPIO 9
TFT_D4   = GPIO 10
TFT_D5   = GPIO 11
TFT_D6   = GPIO 12
TFT_D7   = GPIO 13

Font GLCD   loaded
Font 2      loaded
Font 4      loaded
Font 6      loaded
Font 7      loaded
Font 8      loaded
Smooth font enabled

[/code]
Does this mean that the connections are OK and I need to look somewhere else?
Is it normal that the screen is quite bright white as soon as I connect the pico to the computer?
Is it normal that the screen is flickering (so seemingly getting some kind of power through the other connections even when I also disconnect the TFT_RD to 3V3 connection) when I disconnect the 5V line?

**Reply by Bodmer** [reply]
The output looks good.

A few observations:
1. The Pico board is not fully pushed into the breadboard. The black plastic strip of the header should be flush to the board.
2. The UNO board pins on the data side are not spaced on a 0.1" pitch. The gap between LCD_D7 and LCD_D0 is 0.15", so it must have been forced in, bending the pins and possibly damaging the internal contact strips of the breadboard. I use leads with socket ends and no breadboard to avoid this.
3. Check the board vendors page to make  sure the driver chip is an ILI9486, some similar looking boards are fitted with different controllers.

**Reply by fbeleznay** [reply]
It was indeed the breadboard, thanks for your help. All works now.
![GameOfLife](https://github.com/Bodmer/TFT_eSPI/assets/111730066/e312a239-3942-4115-aaec-79072ad395f2)

**Reply by Bodmer** [reply]
Great to see you have it working, have fun with it.
